OOP/LibraryManagement/LibraryApp/Repositories/JsonFileDocumentRepository.cs

```diff
@@ -0,0 +1,99 @@
+﻿using LibraryApp.Documents;
+using Newtonsoft.Json.Linq;
+using System.Text.RegularExpressions;
+using System.Text.Json;
+using LibraryApp.Services;
+
+namespace LibraryApp.Repositories;
+
+public class JsonFileDocumentRepository<T>(DocumentCache cache) : IRepository<T> where T : Document
+{
+    private const string DocType = "documentType";
+    private readonly DocumentCache _cache = cache;
+    private readonly string _baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
+    private readonly JsonSerializerOptions _jsonOptions = new() { WriteIndented = true, IncludeFields = true };
+
+    public T Read(string title)
+    {
+        var document = _cache.GetDocument<T>(title);
+
+        if (document == null)
+        {
+            var file = new DirectoryInfo(_baseDirectory).GetFiles().FirstOrDefault(f => f.Name.Contains(title));
+
+            if (file != null)
+            {
+                var json = File.ReadAllText(file.FullName);
+                var jObject = JObject.Parse(json);
+                var docType = jObject[DocType].ToObject<DocumentType>();
+
+                document = Deserialize(jObject, docType);
+                _cache.SetDocument(document);
+            }
+        }
```

Comment:
Suggest to reduce extra nesting since it makes code a little bit difficult to read and extend. So, it should look like this:
```suggestion
        if (document != null)
        {
            return document;
         }
         
          var file = new DirectoryInfo(_baseDirectory).GetFiles().FirstOrDefault(f => f.Name.Contains(title));
   
          if (file == null)
          {
                return null;
          }
          
          var json = File.ReadAllText(file.FullName);
          var jObject = JObject.Parse(json);
          var docType = jObject[DocType].ToObject<DocumentType>();

          document = Deserialize(jObject, docType);
           _cache.SetDocument(document);
        }
```